### PR TITLE
AITT-817 part 2: add new statuses to rubric student selector

### PR DESCRIPTION
**Pull request by davidsbailey**
Depends on https://github.com/code-dot-org/code-dot-org/pull/61667. Finishes [AITT-817: Alert teachers of waiting scores - US 4](https://codedotorg.atlassian.net/browse/AITT-817). 

### user-visible changes

statuses on initial load:
| level status | last attempt evaluated? | has teacher feedback? | status |
|--------|--------|--------|--------|
| not attempted | no |  no | not started |
| attempted | no | no | in progress |
| submitted | no | no | submitted | 
| * | yes | no | ready to review |
| * | * | yes | evaluated | 

status changes on user actions:
- teacher evaluates project → ready to review
- teacher evaluates entire class → ready for review
- teacher submits feedback → evaluated

### updates to RubricContainer state

| old name | new name | description |
|--------|--------|--------|
| allAiEvaluationStatus | aiEvalStatusCounters | counters consumed by Run AI Assessment for Class button in RubricSettings |
| - | aiEvalStatusMap | map from student user to ai evaluation status (Not started, In progress, Ready for review). consumed by StudentSelector changes in this PR |
| allTeacherEvaluationData | - | list of teacher evaluation data for each student. consumed by Download CSV button in RubricSettings |
| - | hasTeacherFeedbackMap | map from student to whether they have any teacher feedback. consumed by StudentSelector changes in this PR | 

#### before
student-selector-before

#### after
student-selector-after

## Testing story
* unit tests cover:
  * 5/5 initial statuses
  * 1/3 state transitions: teacher evaluates project → ready to review
* ui tests cover:
  * 2/5 initial statuses: In progress, Ready to review
  * 3/3 state transitions

## Future work

* **move state from RubricContainer into redux.** The states in these components are getting a bit unwieldy. given that the next feature request [AITT-816: Alert teachers of waiting scores - US 3](https://codedotorg.atlassian.net/browse/AITT-816) requires adding a bunch more state, and the state will need to update without reloading the page, now seems like a good time to add a redux store.

### CODE REVIEW

File: apps/src/templates/rubrics/RubricContainer.jsx [hunk 1]
```diff
@@ -134,17 +136,49 @@ export default function RubricContainer({
...
+    const hasFeedbackMap = {};
+    allTeacherEvaluationData.forEach(userEvalData => {
+      if (userEvalData?.user_id) {
+        hasFeedbackMap[userEvalData.user_id] = userEvalData.eval.length > 0;
+      }
+    });
+    setHasTeacherFeedbackMap(hasFeedbackMap);
+  };
+
+  const updateHasTeacherFeedback = userId => {
```
Comment (etaderhold) [inline apps/src/templates/rubrics/RubricContainer.jsx]: nit: the name of this function makes me assume the value of `hasTeacherFeedback` could be updated to any value, but it actually just sets the value to true for this user. Maybe `onSubmitTeacherFeedback` would better reflect the fact that it's a callback to be executed when feedback is submitted.

File: apps/src/templates/rubrics/RubricContainer.jsx [hunk 2]
```diff
@@ -397,7 +434,8 @@ export default function RubricContainer({
               tabSelectCallback={tabSelectCallback}
               reportingData={reportingData}
               allTeacherEvaluationData={allTeacherEvaluationData}
-              allAiEvaluationStatus={allAiEvaluationStatus}
+              aiEvalStatusCounters={aiEvalStatusCounters}
+              updateAiEvalStatusMap={updateAiEvalStatusMap}
```
Comment (etaderhold) [inline apps/src/templates/rubrics/RubricContainer.jsx]: Wouldn't this work just as well and remove the need for the wrapper function above?

```suggestion
              updateAiEvalStatusMap={setAiEvalStatusMap}
```

**Review by etaderhold — APPROVED** [review]
Agreed that adding redux seems like a fine next step given the amount of state (and update functions!) being passed all over the place.

File: apps/src/templates/rubrics/RubricContainer.jsx (hunk 1 above)
Comment (davidsbailey) [reply apps/src/templates/rubrics/RubricContainer.jsx]: works for me!

**Pull request merged**